.github/workflows/build-image.yaml

```diff
@@ -0,0 +1,74 @@
+name: Build Image
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+  push:
+    branches:
+      - main
+      - release/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build-image:
+    name: Build and publish uptime-checker to GHCR
+    if: !github.event.pull_request.head.repo.fork && github.actor != 'dependabot[bot]'
+    runs-on: ubuntu-latest
+
+    env:
+      VERSIONED_IMAGE: ghcr.io/getsentry/uptime-checker:${{ github.sha }}
+      NIGHTLY_IMAGE: ghcr.io/getsentry/uptime-checker:nightly
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+
+      - name: Login to GHCR
+        run: docker login --username '${{ github.actor }}' --password '${{ secrets.GITHUB_TOKEN }}' ghcr.io
+
+      - name: Build and push to GHCR
+        run: |
+          docker buildx build \
+              --platform linux/arm64,linux/amd64 \
+               $( [[ "${GITHUB_REF}" == "refs/heads/main" ]] && printf %s "--tag ${NIGHTLY_IMAGE}" ) \
+              --tag "${VERSIONED_IMAGE}" \
+              --file Dockerfile
+              --push
+              .
+
+  publish-to-dockerhub:
+    name: Publish uptime-checker to DockerHub
```

Comment:
After publishing to GHCR, you could remove the GCR build and just copy the GHCR image to GCR like what Relay did here https://github.com/getsentry/relay/blob/ac330795fdb9d00ce158f200c380b7b364100bd5/.github/workflows/ci.yml#L403-L408

Reply:
All this should be in a separate patch

Reply:
@BYK By "all this", did you mean removing the GCR build?

Reply:
Yup, this entire job.

Reply:
I don't see why not now? If we rely on the GCR build, we would need the `until ...` command that would take up a long time for the GHA to finish.